tox.ini

```diff
@@ -133,7 +133,9 @@ commands_pre=
     pip install -r {envtmpdir}/requirements.txt
     pip freeze
 commands=
-    pytest astropy/astropy/io/misc/asdf --open-files --run-slow --remote-data -W "ignore:The property AsdfFile.blocks has been deprecated:asdf.exceptions.AsdfDeprecationWarning:asdf.asdf:661"
+    pytest astropy/astropy/io/misc/asdf --open-files --run-slow --remote-data \
+        -W "ignore:The property AsdfFile.blocks has been deprecated:asdf.exceptions.AsdfDeprecationWarning:asdf.asdf:661" \
+        -W "ignore::asdf.exceptions.AsdfDeprecationWarning:asdf.types:504"
```

Comment:
This will be fixed by the above request on the entry point.

Reply:
This is still needed as the tests fail on collection without it. Here is one example:
```
plugins: remotedata-0.4.0, xdist-3.2.0, mock-3.10.0, filter-subpackage-0.1.2, astropy-header-0.2.2, doctestplus-0.12.1, hypothesis-6.68.1, astropy-0.10.0, asdf-2.14.4.dev225+gf580ac36.d20230215, cov-4.0.0, openfiles-0.5.0, arraydiff-0.5.0
collected 289 items / 25 errors

====================================================================================================================== ERRORS =======================================================================================================================
________________________________________________________________________________________________ ERROR collecting astropy/io/misc/asdf/extension.py _________________________________________________________________________________________________
astropy/astropy/io/misc/asdf/extension.py:12: in <module>
    from .tags.coordinates.angle import *
astropy/astropy/io/misc/asdf/tags/coordinates/angle.py:3: in <module>
    from astropy.io.misc.asdf.tags.unit.quantity import QuantityType
astropy/astropy/io/misc/asdf/tags/unit/quantity.py:4: in <module>
    from astropy.io.misc.asdf.types import AstropyAsdfType
astropy/astropy/io/misc/asdf/types.py:37: in <module>
    class AstropyType(CustomType, metaclass=AstropyTypeMeta):
astropy/astropy/io/misc/asdf/types.py:25: in __new__
    cls = super().__new__(mcls, name, bases, attrs)
../../../asdf/types.py:122: in __new__
    new_cls = super().__new__(cls, name, bases, attrs)
../../../asdf/types.py:504: in __init_subclass__
    warnings.warn(
E   asdf.exceptions.AsdfDeprecationWarning: AstropyType from astropy.io.misc.asdf.types subclasses the deprecated CustomType class. Please see the new extension API https://asdf.readthedocs.io/en/stable/asdf/extending/converters.html
```